core/src/test/java/mediumtest/ConnectionSetupMediumTests.java

```diff
@@ -202,6 +204,44 @@ void it_should_fails_to_validate_connection_if_host_not_found() throws IOExcepti
     assertThat(connectionResponse.getMessage()).contains("notexists");
   }
 
+  @Test
+  void it_should_support_notifications_if_sonarcloud() throws ExecutionException, InterruptedException {
+    var fakeClient = newFakeClient().build();
+    backend = newBackend().build(fakeClient);
+
+    var connectionResponse = backend.getConnectionService()
+      .checkSmartNotificationsSupported(new CheckSmartNotificationsSupportedParams(
+        new TransientSonarCloudConnectionDto("https://sonarcloud.io", Either.forLeft(new TokenDto("foo"))))).get();
+
+    assertThat(connectionResponse).isTrue();
+  }
+
+  @Test
+  void it_should_support_notifications_if_sonarqube_9_7() throws ExecutionException, InterruptedException {
+    var fakeClient = newFakeClient().build();
+    server = newSonarQubeServer("9.7").start();
+    backend = newBackend().withEmbeddedServer().withSonarQubeConnection("connectionId", server).build(fakeClient);
+
+    var connectionResponse = backend.getConnectionService()
+      .checkSmartNotificationsSupported(new CheckSmartNotificationsSupportedParams(
+        new TransientSonarQubeConnectionDto(server.baseUrl(), Either.forLeft(new TokenDto("foo"))))).get();
+
+    assertThat(connectionResponse).isTrue();
+  }
+
+  @Test
+  void it_should_not_support_notifications_if_sonarqube_7_9() throws ExecutionException, InterruptedException {
```

Comment:
To make it more clear what the limit is, you could have a test with a version just before they were supported, and another test with the version that brings support.

Reply:
I decided to go for the manual `withSmartNotificationsSupported(boolean)` as there are a lot of cases to take into account if we were to test between edition and version